src/uwtools/tests/test_cli.py

```diff
@@ -237,23 +253,37 @@ def test__dispatch_config_translate_arparse_to_jinja2():
     convert.assert_called_once_with(input_file=1, output_file=3, dry_run=5)
 
 
+def test__dispatch_config_translate_no_optional():
+    args, _ = cli._parse_args("uw config translate".split()[1:])
+    with patch.object(cli.uwtools.config.atparse_to_jinja2, "convert") as convert:
+        cli._dispatch_config_translate(args)
+    convert.assert_called_once_with(input_file=None, output_file=None, dry_run=False)
+
+
 def test__dispatch_config_translate_unsupported():
     args = {STR.infile: 1, STR.infmt: "jpg", STR.outfile: 3, STR.outfmt: "png", STR.dryrun: 5}
     assert cli._dispatch_config_translate(args) is False
 
 
-def test__dispatch_config_validate_yaml():
-    args = {STR.infile: 1, STR.infmt: FORMAT.yaml, STR.schemafile: 3}
+def test__dispatch_config_validate_no_optional():
+    args, _ = cli._parse_args("uw config validate --schema-file /foo.schema".split()[1:])
```

Comment:
The issue here appears to be [this conditional](https://github.com/ufs-community/workflow-tools/blob/ac5879ad2c863d57f7966532d1a0056e9a7facdb/src/uwtools/cli.py#L227C5-L227C5) on the file format. I'm not sure why `args[STR.infmt]` is not working but `args.input_format` worked before. 
For that matter, do we even want it to be confirming the format, or just accepting an override? If we do, then it should be required if input_file is provided, or perform the type check for the user instead of taking the option

Reply:
The line of code you've linked to is in `_dispatch_config_translate()`, not `_dispatch_config_validate()` under test here. Do you mean [this](https://github.com/ufs-community/workflow-tools/blob/ac5879ad2c863d57f7966532d1a0056e9a7facdb/src/uwtools/cli.py#L245)?

If so, it seems to me like we have to check that the format is YAML before we call `uwtools.config.validator.validate_yaml()`. That's currently the only data format we have a schema for (except Rocoto XML, but it's got its own CLI [sub]mode -- we could also support it here, but that's a question for another day).

Although the input-format CLI argument is technically optional from the perspective of `argparse`, there's a post-parse check that is supposed to ensure that, if an input file wasn't specified (i.e. we are reading `stdin`), the input format **is**, as it will otherwise be impossible to deduce the format from a non-existent filename extension.

Do you have an idea for changing this? I'm not sure I understood if you were proposing something.

Reply:
AH sorry, you're right - I think this was my misunderstanding of everything `_check_file_vs_format` was doing and how we were breaking the format variable before. Its already doing my proposal of making it check the format if not provided, but it has to take a real path not a mocked `None`